File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/projects/l2-resource-for-range/submodule.ts
import * as pulumi from "@pulumi/pulumi";
import * as simple from "@pulumi/simple";

interface SubmoduleArgs {
    submoduleListVar: pulumi.Input<string[]>,
    submoduleFilterCond: pulumi.Input<boolean>,
    submoduleFilterVariable: pulumi.Input<number>,
}

export class Submodule extends pulumi.ComponentResource {
    constructor(name: string, args: SubmoduleArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:Submodule", name, args, opts);
        const submoduleRes: simple.Resource[] = [];
pulumi.output(args.submoduleListVar).apply(toOutput => toOutput.map((v, k) => [k, v]).filter(([k, v]) => pulumi.output(args.submoduleFilterCond)).reduce((__obj, [k, v]) => ({ ...__obj, [k]: v }))).apply(rangeBody => {
            for (const range of Object.entries(rangeBody).map(([k, v]) => ({key: k, value: v}))) {
                submoduleRes.push(new simple.Resource(`${name}-submoduleRes-${range.key}`, {value: true}, {
                parent: this,
            }));
            }
        });

        const submoduleResWithApplyFilter: simple.Resource[] = [];
pulumi.all([pulumi.output(args.submoduleListVar), pulumi.output(args.submoduleFilterVariable)]).apply(([toOutput, toOutput1]) => toOutput.map((v, k) => [k, v]).filter(([k, v]) => toOutput1 == 1).reduce((__obj, [k, v]) => ({ ...__obj, [k]: v }))).apply(rangeBody => {
            for (const range of Object.entries(rangeBody).map(([k, v]) => ({key: k, value: v}))) {
                submoduleResWithApplyFilter.push(new simple.Resource(`${name}-submoduleResWithApplyFilter-${range.key}`, {value: true}, {
                parent: this,
            }));
            }
        });

        this.registerOutputs();
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/projects/l2-resource-for-range/submodule.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_simple as simple

class SubmoduleArgs(TypedDict, total=False):
    submoduleListVar: Input[list[str]]
    submoduleFilterCond: Input[bool]
    submoduleFilterVariable: Input[int]

class Submodule(pulumi.ComponentResource):
    def __init__(self, name: str, args: SubmoduleArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:Submodule", name, args, opts)

        submodule_res = []
def create_submodule_res(range_body):
            for range in [{"key": k, "value": v} for [k, v] in enumerate(range_body)]:
                submodule_res.append(simple.Resource(f"{name}-submoduleRes-{range['key']}", value=True,
                opts = pulumi.ResourceOptions(parent=self)))

pulumi.Output.from_input(args["submoduleListVar"]).apply(lambda resolved_outputs: create_submodule_res(        {k: v for k, v in resolved_outputs['to_output'] if pulumi.Output.from_input(args["submoduleFilterCond"])}))

        submodule_res_with_apply_filter = []
def create_submodule_res_with_apply_filter(range_body):
            for range in [{"key": k, "value": v} for [k, v] in enumerate(range_body)]:
                submodule_res_with_apply_filter.append(simple.Resource(f"{name}-submoduleResWithApplyFilter-{range['key']}", value=True,
                opts = pulumi.ResourceOptions(parent=self)))

pulumi.Output.all(
            to_output=pulumi.Output.from_input(args["submoduleListVar"]),
            to_output1=pulumi.Output.from_input(args["submoduleFilterVariable"])
).apply(lambda resolved_outputs: create_submodule_res_with_apply_filter(        {k: v for k, v in resolved_outputs['to_output'] if resolved_outputs["'to_output1'"] == 1}))

        self.register_outputs()
